src/client/auth/actions.js

```diff
@@ -33,14 +33,23 @@ function validateForm(fields) {
 }
 
 function validateCredentials(fields) {
-  // Simulate lengthy async action.
   return new Promise((resolve, reject) => {
-    setTimeout(() => {
-      if (fields.password === 'pass1')
+
+    var xhr = new XMLHttpRequest();
```

Comment:
Yeah, I agree. Was looking at this https://github.com/github/fetch a while ago. Have you tried both of them? Any opinions/issues?